sim/core/aura.go

```diff
@@ -18,6 +18,7 @@ type OnInit func(aura *Aura, sim *Simulation)
 type OnReset func(aura *Aura, sim *Simulation)
 type OnDoneIteration func(aura *Aura, sim *Simulation)
 type OnGain func(aura *Aura, sim *Simulation)
+type OnRefresh func(aura *Aura, sim *Simulation)
```

Comment:
New callback used to handle some effects that we can't quite handle through a buff aura, e.g. Shield Mastery restoring mana on apply/refresh